### PR TITLE
Imports and Contraband 2: Landfill Gacha Addiction (I put trash randomizers into cargo crates and called it content)

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

This is a followup on my previous PR involving cargo imports. I've made a number of changes and new additions to cargo's imports and contraband. But I've also changed how Smuggler's Satchels generate loot as well.

### New:
**Abandoned Crates:** You can now order in abandoned crates at a steep price. Obviously these are just your standard fare abandoned crates, so they've got a pretty long list of potential contents. Some great, some utterly not worth the price you paid for the crate. Since they're quite pricey, you can't order very many quickly. But this does allow cargo techs the opportunity to spend the round solving puzzles to get interesting loot.

**Dumpster of Maint Garbage:** This dumpster (similarly named to another dumpster you can order) is filled with maint trash and potential maint random spawns. This list is extensive enough that anything spawned in this crate is likely to be mostly garbage. But, it is more affordable than abandoned crates. I'd consider this the literally trashier version of the abandoned crate.

**Shamber's Juice Eldritch Energy! Crate:** A crate with one can of the extremely rare, short run edition of Shambler's Juice, Eldritch Energy! This contains 5 units of eldritch essence. Heals heretics, hurts everyone else! This is a VERY potent poison, but it also happens to be a handy way for a Cargonian heretic to get a potent healing item without having to waste knowledge points.

**Animal Hide Crate:** It's a cargo crate full of animal hides! This can include fairly rare hides and some icebox creature hides as well, like polar bear hides and wolf sinew. It's not too expensive, and mostly spits out leather.

**Dreadnog Carton Crate:** A carton full of the worst eggnog imaginable. This is just something to troll people with. Drink it and you'll get a massive mood penalty. Dreadnog! May or may not contain space cola!

### Updated:

**Contraband Crate and Smuggler's Satchels:** This has had it's price increased considerably. But, for good reason. It now contains some more controlled random items, but also some more valuable contraband as well as a very rare spawn. The upper end on his contraband can be extremely valuable, but the majority of the items gained from contraband crates will probably be either what you can get now (quite weak), or something a bit more middle of the road (some more unique narcotics).

As a consequence, I've also passed this change onto smuggler's satchels, as they used the crate to generate its contents. (it literally spawned and then deleted a contraband crate to generate the contents hoo haa). 

I've also increased the number of items in the smuggler's satchel. Since the randomly spawned smuggler's satchels are quite a bit rarer now there is only ever two spawned in the map, and spending actual TC on these is somewhat counterproductive, I don't imagine this will be more beneficial for scavenger hunters hoping for some interesting goodies.

**Russian Crate (the normal one):** The mosins now spawn in ancient gun cases. These determine what kind of mosin you can get. 79% of the time, you get the crap mosin. 20% of the time, you get a good mosin. And 1% of the time, you get rations. This more tightly controls how many good mosins are entering into the round and how much of a value purchase the Russian crate actually is for getting ballistics. Since the process is even more unlikely than before, it isn't necessarily as guaranteed that you will get a good mosin. Hell, you might not even get a gun if you're that unlucky.

**Shocktrooper Crate:** It now has an armor vest and helmet. So, not only do you get some grenades, you get some protection as well. Since this is the 'loud' crate, I felt it appropriate to make it slightly more useful for enabling that.

**Special Ops Crate:** It now contains five mirage grenades and a chameleon belt, and has had the survival knife improved to a switchblade. This is probably the weakest of the two crates STILL, but hopefully these make them a little more interesting and novel by giving them pretty fun grenade to toy with.

## Why It's Good For The Game

My initial PR hoped to add in a few more interesting purchases for cargo. I think currently cargo has a slight issue of not having enough valuable or interesting uses for their money. I think it still has that problem, but by including more unique crates that allow cargo to provide some oddities into the round, that might slowly work itself out.

This PR hopes to provide another way to waste their money if they have an excess amount. Landfill Trash Gambling. Spending it away on complete junk, which I think is absolutely hilarious when it doesn't work out, as it is soulful in its design. Definitely not inspired by my recent thrift shop excursions this month buying and scrounging for furniture and interesting clothing. 

[Relevant](https://www.youtube.com/watch?v=QK8mJJJvaes)

Also, I wanted to buff some of the crates I introduced a bit last time, and nerf the mosin production somewhat via a more controllable method that I can actually adjust as necessary down the line.

## Changelog
:cl:
fix: Stops manifest generation runtiming when a cargo crate is empty.
add: Abandoned crates are now available via cargo imports.
add: Dumpsters full of maintenance trash are now available via cargo imports.
add: An ultra-rare can of Shambler's Juice is now available via cargo imports.
add: Animal hides and leathers can be (unreliably) ordered via cargo imports.
add: The Dreadnog has entered this realm. To consume, purchase it via cargo imports.
balance: Contraband Crates (and as a consequence, smuggler's satchels) now generate more varied goods. Mostly the same, but sometimes you get something quite different or even valuable.
balance: Mosins generated via the Russian supply crate are a bit more random, weighing more heavily towards bad mosins than good mosins.
balance: Buffed both the shocktrooper and special op crate. Shocktrooper now has an armored helmet and vest, and special op now has 5 mirage grenades and a chameleon belt. The survival knife in the special op crate is now a switchblade.
/:cl:

### CODE REVIEW

**Comment by YuliceTheDruid** [comment]
Neat ideas, lot of crates too expensive. Dreadnog is going to be prime grief tool unfortunately as funny as it is.

**Comment by necromanceranne** [comment]
Prices are subject to change in every instance. I'd rather price high and reduce later than create an issue for people in the rounds while it is priced too low.

**Comment by carlarctg** [comment]
If you want some more inspiration for simple and useful Cargo expansion, Goodies could use an overhaul, especially some way of more easily sorting through the items. Additionally there's as usual lots of neat small trinkets that can be slapped on there. For example the 'pair pinpointer' that tracks a bound pinpointer

I say this a little because while I like this PR you're trying to fix the issue of cargo being barren by adding on more 'illegal contraband' items, which is probably not the only way cargo should be able to purchase interesting things.

**Comment by carlarctg** [comment]
> Prices are subject to change in every instance. I'd rather price high and reduce later than create an issue for people in the rounds while it is priced too low.

I think this should definitely be the opposite. It's better to make something 'overpowered' than 'underpowered', because OP things are hammered into shape, and UP things languish forever. For example, the MEAT crate. It's fun, but it's so costly and only gives one MEAT that nobody ever buys the crate.

File: code/datums/mood_events/drink_events.dm [hunk 1]
```diff
@@ -18,6 +18,11 @@
 	mood_change = -2
 	timeout = 6 MINUTES
 
+/datum/mood_event/quality_revolting
+	description = "That drink was the worst thing I've ever consumed."
```
Comment (CRITAWAKETS) [inline code/datums/mood_events/drink_events.dm:22]: alternate description: I hated that drink! It was revolting!

**Comment by necromanceranne** [comment]
> If you want some more inspiration for simple and useful Cargo expansion, Goodies could use an overhaul, especially some way of more easily sorting through the items. Additionally there's as usual lots of neat small trinkets that can be slapped on there. For example the 'pair pinpointer' that tracks a bound pinpointer

I'll look at goodies a bit later, but for now I just want to follow up on the imports category. I'd love to give some more love to goodies, even though I've definitely touched it before in a few past prs.

> I say this a little because while I like this PR you're trying to fix the issue of cargo being barren by adding on more 'illegal contraband' items, which is probably not the only way cargo should be able to purchase interesting things.

It's not _entirely_ illegal. I'm more interested in adding curiosities if anything as a kind of odd luxury item.

And to save me a quote, I understand overshooting and working back would be ideal for how I would like to proceed, but I've been yelled at by angry people whether I undershoot or overshoot, and I find it easier to work from undershooting and getting less sideways looks from maints than I do overshooting and getting a bunch of people furious from all sides.

**Comment by CRITAWAKETS** [comment]
I'm not sure if a maint spawner crate is exactly a good idea, since it means a theoretical infinite amount of chances to roll for rare maint items. Honestly the issue isn't the actual quality, maint items suck, but I'm not sure if it's a thematically appropriate thing for it to be buyable.

**Comment by necromanceranne** [comment]
> I'm not sure if a maint spawner crate is exactly a good idea, since it means a theoretical infinite amount of chances to roll for rare maint items. Honestly the issue isn't the actual quality, maint items suck, but I'm not sure if it's a thematically appropriate thing for it to be buyable.

It's something I'm very much um'ing about. I mostly want impressions first about some of the inclusions. The quality is what mostly makes me think it should be okay, and I think framing it as 'someone is sending you literal garbage for a premium' is kind of in-keeping with the dubious value of the imports section.

**Comment by nukularpower** [comment]
This is great in my opinion.  Was just saying I wished there was SOME other way to get lead (for Saturn-X.)   Hopefully the maint crates contain a decent amount of lead pipes/batteries.

**Comment by EricZilla** [comment]
I think it would be funny if the maint garbage was labeled "Divine Heavens Gatchapon Box" or some other mabinogi ass bs

**Comment by CCC23** [comment]
YEAA!! MORE CARGO STUFFF!!!

**Comment by nukularpower** [comment]
> > This is great. Was just saying I wished there was SOME other way to get lead (for Saturn-X.) Hopefully the maint crates contain a decent amount of lead pipes/batteries.
> 
> Just mutate plums into plumb

if thats actually true, its not on the wiki :(

**Pull request merged**